api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java

```diff
@@ -641,6 +651,31 @@ public void testGetAllUsersWithActiveInitialCredits() {
     assertThat(activeUsers).doesNotContain(expiredAndCleanedUser);
   }
 
+  @Test
+  public void testGetAllUserIdsWithCurrentTierAccess() {
+    DbUser rtUser = userDao.save(new DbUser().setUserId(1).setUsername("rt"));
+    DbUser exRtUser = userDao.save(new DbUser().setUserId(2).setUsername("ex-rt"));
+    DbUser ctUser = userDao.save(new DbUser().setUserId(3).setUsername("ct"));
+    DbUser exCtUser = userDao.save(new DbUser().setUserId(4).setUsername("ex-ct"));
+    DbUser noTierUser = userDao.save(new DbUser().setUserId(5).setUsername("none"));
+
+    // add "rt" and "ex-rt" to the Registered Tier, but then remove ex-rt
+
+    accessTierService.addUserToTier(rtUser, registeredTier);
+    accessTierService.addUserToTier(exRtUser, registeredTier);
+    accessTierService.removeUserFromTier(exRtUser, registeredTier);
```

Comment:
note that adding and removing is a different state in the DB from never having access at all.